lib/internal/child_process.js

```diff
@@ -453,15 +453,23 @@ function setupChannel(target, channel) {
       //Linebreak is used as a message end sign
       while ((i = jsonBuffer.indexOf('\n', start)) >= 0) {
         var json = jsonBuffer.slice(start, i);
-        var message = JSON.parse(json);
+        var message;
 
-        // There will be at most one NODE_HANDLE message in every chunk we
-        // read because SCM_RIGHTS messages don't get coalesced. Make sure
-        // that we deliver the handle with the right message however.
-        if (message && message.cmd === 'NODE_HANDLE')
-          handleMessage(target, message, recvHandle);
-        else
-          handleMessage(target, message, undefined);
+        try {
```

Comment:
I'm not sure if this is the case in the version of V8 that ships with v7, but this `try...catch` would deoptimize this whole function.